frontend/src/features/mapSideBar/SidebarContents/AssociatedLandForm.tsx

```diff
@@ -340,15 +346,20 @@ const AssociatedLandForm: React.FC<IAssociatedLandParentForm> = (
     let errors = await yupErrors;
     await ownedParcels.forEach(async (p: any, index: number) => {
       let pidDuplicated = false;
-      if (p.pid && getIn(initialValues.data, `parcels.${index}.pid`) !== p.pid) {
+      if (
+        p.pid &&
+        (getIn(initialValues.data, `parcels.${index}.pid`) !== p.pid ||
+          !getIn(initialValues.data, `parcels.${index}.id`))
```

Comment:
if we have an id then this will be treated as an update, so no need to worry about pid/pin duplicates.